2020/11/index.js

```diff
@@ -1,3 +1,223 @@
+const { promises: fs } = require('fs');
+
+async function getInput() {
+  return input = (await fs.readFile(`${__dirname}/input.txt`))
+    .toString()
+    .trim()
+    .split('\n')
+    .map(row => row.split(''));
+}
+
+function p1NumAdjOccupiedSeats(input, row, column) {
+  const rows = [row];
+  if(row - 1 >= 0) {
+    rows.push(row - 1);
+  }
+  if(row + 1 < input.length) {
+    rows.push(row + 1);
+  }
+
+  const columns = [column];
+  if(column - 1 >= 0) {
+    columns.push(column - 1);
+  }
+  if(column + 1 < input[0].length) {
+    columns.push(column + 1);
+  }
+
+  let numAdjOccupied = 0;
+
+  for(const i of rows) {
+    for(const j of columns) {
+      if(i === row && j === column) {
+        continue;
+      }
+
+      numAdjOccupied += input[i][j] === '#'
+        ? 1
+        : 0;
+    }
+  }
+
+  return numAdjOccupied;
+}
+function p2NumAdjOccupiedSeats(input, row, column) {
+  /*
+   * x x x      0 1 2
+   * x x x  =>  3 4 5
+   * x x x      6 7 8
+   */
+  const adj = {
+    0: { row: row - 1, column: column - 1 },
+    1: { row: row - 1, column },
+    2: { row: row - 1, column: column + 1 },
+    3: { row: row, column: column - 1 },
+    5: { row: row, column: column + 1 },
+    6: { row: row + 1, column: column - 1 },
+    7: { row: row + 1, column: column },
+    8: { row: row + 1, column: column + 1 },
+  };
+
+  const maxRows = input.length;
+  const maxColumns = input[0].length;
+  let numAdjOccupied = 0;
+
+  function shouldContinue(key) {
+    if(!adj[key]) {
+      return false;
+    }
+
+    const { row, column } = adj[key];
+
+    if(row < 0 || column < 0 || row === maxRows || column === maxColumns ) {
+      delete adj[key];
```

Comment:
and you lost me!

Reply:
Instead of checking for negative values, I just removed that direction from being considered in the next iteration of checks.